### DISCUSSION TITLE
Excluding binaries

### DISCUSSION THREAD

**Discussion by Piotr-Miller** [SteveGilham/altcover#130] [Q&A] [answered]
Hi Guys,

Got question about excluding particular binary from instrumentation, let me call it xxx.dll.
This is not test binary, but I want to exclude it anyway.
I was trying different approaches, but it seems none of them are working, for instance:

altcover --inplace -i "path to release folder" -s="xxx.dll"
altcover --inplace -i "path to release folder" -e="xxx.dll"

I will appreciate any tip.

**Comment by SteveGilham** [comment, marked as answer]
Those filters work on the assembly name (the value used for linkage) rather than the assembly path (which includes the file name extension).  Strictly, the matched string is `System.Reflection.AssemblyName.GetAssemblyName(path to assembly).Name`; but in all reasonable cases that is the same as the value of `System.IO.Path.GetFileNameWithoutExtension(path to assembly)`.

The string supplied as a filter argument is a regex that is matched against the assembly name; that also means that a literal `.` in the filter is a wild-card match --  those examples are excluding assemblies like `path to release folder\xxx0dll.dll` or `path to release folder\xxxZdll.exe`

The command line you want would be
```
altcover --inplace -i "path to release folder" -s="yyy"
```
where `yyy` is some unique substring of the assembly name, regex escaped.  As an example, from the AltCover build process
```
let AltCoverFilter (p: Primitive.PrepareOptions) =
    { p with
          AssemblyFilter =
              [ @"\.DataCollector"; "Sample" ]
              @ (p.AssemblyFilter |> Seq.toList)
...
```
which excludes the `AltCover.DataCollector` assembly, and all the code used as test data in assemblies called `Sample##.dll` for various values of `##`.

**Reply by Piotr-Miller** [reply]
Thanks Steve for explanation. It's working as a charm.
